pkg/controller/statefulset/stateful_set_utils_test.go

```diff
@@ -937,3 +938,36 @@ func newStatefulSetWithLabels(replicas int, name string, uid types.UID, labels m
 		},
 	}
 }
+
+func TestGetStatefulSetMaxUnavailable(t *testing.T) {
+	testCases := []struct {
+		maxUnavailable         *intstr.IntOrString
+		replicaCount           int
+		expectedMaxUnavailable int
+	}{
+		{maxUnavailable: nil, replicaCount: 10, expectedMaxUnavailable: 1},
+		{maxUnavailable: intOrStrP(intstr.FromInt(3)), replicaCount: 10, expectedMaxUnavailable: 3},
+		{maxUnavailable: intOrStrP(intstr.FromInt(3)), replicaCount: 0, expectedMaxUnavailable: 3},
+		{maxUnavailable: intOrStrP(intstr.FromString("10%")), replicaCount: 25, expectedMaxUnavailable: 2},
+		{maxUnavailable: intOrStrP(intstr.FromString("100%")), replicaCount: 5, expectedMaxUnavailable: 5},
+		{maxUnavailable: intOrStrP(intstr.FromString("50%")), replicaCount: 5, expectedMaxUnavailable: 2},
+		{maxUnavailable: intOrStrP(intstr.FromString("10%")), replicaCount: 5, expectedMaxUnavailable: 1},
+		{maxUnavailable: intOrStrP(intstr.FromString("1%")), replicaCount: 0, expectedMaxUnavailable: 1},
```

Comment:
hm, I suppose it wouldn't hurt to also test the `0` and `0%` even with this comment in mind

```
// Note that API validation has already guaranteed the maxUnavailable field to be >1 if it is an integer
// or 0% < value <= 100% if it is a percentage, so we don't have to consider other cases.
```

Reply:
no pro, added :)